src/overlays/actors/ovl_Boss_Ganon2/z_boss_ganon2.c

```diff
@@ -278,116 +304,3125 @@ static ColliderJntSphInit D_80907014 = {
+        phi_f20.x = D_80910608[phi_s2].x - sp2D0.x;
+        phi_f20.y = D_80910608[phi_s2].y - sp2D0.y;
+        phi_f20.z = D_80910608[phi_s2].z - sp2D0.z;
+        if (phi_s2 < 3) {
+            if (sqrtf(SQXYZ(phi_f20)) <= phi_f30) {
+                phi_s2++;
+                phi_f30 = sp294;
+            }
+        } else {
+            if (sqrtf(SQXYZ(phi_f20)) <= (temp_f28 + 1.0f)) {
+                phi_s2++;
+            }
+        }
+        temp_s4_2++;
+        if ((temp_s4_2 < 20) && (phi_s2 < 4)) {
+            phi_f20.x = D_80910608[phi_s2].x - sp2D0.x;
+            phi_f20.y = D_80910608[phi_s2].y - sp2D0.y;
+            phi_f20.z = D_80910608[phi_s2].z - sp2D0.z;
+        } else {
+            break;
+        }
+    }
+
+    temp_a1 = SEGMENTED_TO_VIRTUAL(ovl_Boss_Ganon2_Vtx_00BA20);
+    for (i = 0; i < 11; i++) {
+        if ((temp_s4 - i) > 0) {
+            temp_a1[D_80907084[i]].n.ob[0] = sp18C[temp_s4 - i - 1].x;
+            temp_a1[D_80907084[i]].n.ob[1] = sp18C[temp_s4 - i - 1].y;
+            temp_a1[D_80907084[i]].n.ob[2] = sp18C[temp_s4 - i - 1].z;
+        }
+        if ((temp_s4_2 - i) > 0) {
+            temp_a1[D_80907090[i]].n.ob[0] = sp9C[temp_s4_2 - i - 1].x;
+            temp_a1[D_80907090[i]].n.ob[1] = sp9C[temp_s4_2 - i - 1].y;
+            temp_a1[D_80907090[i]].n.ob[2] = sp9C[temp_s4_2 - i - 1].z;
+        }
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_809034E4.s")
+#endif
+
+void func_80903F38(BossGanon2* this, GlobalContext* globalCtx) {
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_boss_ganon2.c", 5083);
+
+    if (this->unk_312 != 0) {
+        func_809034E4(&this->unk_200, &this->unk_20C);
+        D_80907080 = 0xFF;
+    }
+
+    if (D_80910638 >= 4) {
+        gSPSegment(
+            POLY_XLU_DISP++, 0x08,
+            Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, 0, 32, 32, 1, globalCtx->gameplayFrames * 18, 0, 32, 32));
+        gDPPipeSync(POLY_XLU_DISP++);
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, 255, D_80907080);
+        Matrix_Translate(0.0f, 0.0f, 0.0f, MTXMODE_NEW);
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_boss_ganon2.c", 5117),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_XLU_DISP++, ovl_Boss_Ganon2_DL_00BB80);
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_boss_ganon2.c", 5122);
+}
+
+void func_80904108(BossGanon2* this, GlobalContext* globalCtx) {
+    s32 pad;
+
+    if (this->unk_324 > 0.0f) {
+        OPEN_DISPS(globalCtx->state.gfxCtx, "../z_boss_ganon2.c", 5131);
+
+        Matrix_Push();
+        gDPPipeSync(POLY_XLU_DISP++);
+        gSPSegment(POLY_XLU_DISP++, 0x08,
+                   Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, (s32)globalCtx->gameplayFrames, 0, 32, 64, 1,
+                                    (0 - globalCtx->gameplayFrames) * 2, (0 - globalCtx->gameplayFrames) * 8, 32, 32));
```

Comment:
```suggestion
                                    -globalCtx->gameplayFrames * 2, -globalCtx->gameplayFrames * 8, 32, 32));
```
does that not match?